app/javascript/onboarding/components/FollowUsers.jsx

```diff
@@ -175,42 +181,54 @@ export class FollowUsers extends Component {
               </div>
             </header>
 
-            <div data-testid="onboarding-users">
-              {users.map((user) => (
-                <button
-                  data-testid="onboarding-user-button"
-                  type="button"
-                  onClick={() => this.handleClick(user)}
-                  onKeyDown={() => this.handleKeyDown(user)}
-                  className={
-                    selectedUsers.includes(user)
-                      ? 'user content-row selected'
-                      : 'user content-row unselected'
-                  }
-                >
-                  <figure className="user-avatar-container">
-                    <img
-                      className="user-avatar"
-                      src={user.profile_image_url}
-                      alt="profile"
-                    />
-                  </figure>
-                  <div className="user-info">
-                    <h4 className="user-name">{user.name}</h4>
-                    <p className="user-summary">
-                      {he.unescape(user.summary || '')}
-                    </p>
-                  </div>
-                  <button
-                    data-testid="onboarding-user-following-status"
-                    type="button"
-                    className="user-following-status"
+            <fieldset data-testid="onboarding-users">
+              {users.map((user) => {
+                const selected = selectedUsers.includes(user);
+
+                return (
+                  <div
+                    data-testid="onboarding-user-button"
+                    className={
+                      selected
+                        ? 'user content-row selected'
+                        : 'user content-row unselected'
+                    }
                   >
-                    {selectedUsers.includes(user) ? 'Following' : 'Follow'}
-                  </button>
-                </button>
-              ))}
-            </div>
+                    <figure className="user-avatar-container">
+                      <img
+                        className="user-avatar"
+                        src={user.profile_image_url}
+                        alt="profile"
+                      />
+                    </figure>
+                    <div className="user-info">
+                      <h4 className="user-name">{user.name}</h4>
+                      <p className="user-summary">
+                        {he.unescape(user.summary || '')}
+                      </p>
+                    </div>
+                    <label className="relative">
+                      <input
+                        type="checkbox"
+                        checked={selected}
+                        className="user-following-status invisible absolute top-0 bottom-0 right-0 left-0"
```

Comment:
refactor (blocking): `invisible` status causes the checkbox to not be visible in the accessibility tree. Opt for `opacity-0` instead.

Here's a screenshot with the 1st follow checkbox with the `invisible` CSS class applied and `opacity-0` applied to the other checkboxes.

![image](https://user-images.githubusercontent.com/833231/108013103-0062e680-6fd9-11eb-8e16-b17e088e302c.png)

I'm reworking the map JSX at the moment, so don't make this change yet.

Reply:
Refactor (blocking): Rework the mapped JSX to contain more semantic markup and less markup overall.

This is how I envision the users being mapped over.

```jsx
{users.map((user) => {
  const selected = selectedUsers.includes(user);

  return (
    <div
      className={`user content-row ${
        selected ? 'selected' : 'unselected'
      }`}
    >
      <figure className="user-avatar-container">
        <img
          className="user-avatar"
          src={user.profile_image_url}
          alt="profile"
        />
      </figure>
      <section className="user-info">
        <h4 className="user-name">{user.name}</h4>
        <p className="user-summary">
          {he.unescape(user.summary || '')}
        </p>
      </section>
      <label
        aria-label={`Follow ${user.name}`}
        className={`user-following-status crayons-btn${
          selected ? '' : ' crayons-btn--outlined'
        }`}
      >
        <input
          type="checkbox"
          checked={selected}
          className="absolute opacity-0"
          onClick={() => this.handleClick(user)}
        />
        {selected ? 'Following' : 'Follow'}
      </label>
    </div>
  );
})}
```

The `<label />` has an `aria-label` attribute because the label text content , `"Follow"`, is too generic, so I opted for `aria-label={`Follow ${user.name}`}`.

Along with these changes, there would be some required changes to the CSS as well.

In onboarding-modal.scss, the following would no longer be required

```diff
-  input[type='checkbox'].user-following-status + div {
-    margin-left: 32px;
-  }
-
-  input[type='checkbox']:not(:checked).user-following-status + div {
-    @extend .crayons-btn;
-    @extend .crayons-btn--outlined;
-  }
-
-  input[type='checkbox']:checked.user-following-status + div {
-    @extend .crayons-btn;
-  }
```

and the following change would be required in that same file

```diff
.content-row {
  align-items: center;
  color: #202428;
  display: flex;
  padding: 8px 16px;
  text-align: left;
  width: 100%;

-  &:hover {
+  &:hover,
+  &:focus-within {
    background-color: #eef0f1 !important;
  }
}
```

`&:focus-within` will provide the styling to the content row when the checkbox, even though hidden, receives focus.

Curious on your thoughts on these proposed changes @forem/frontend.

Reply:
I think this suggestion makes sense @nickytonline 👍  

It would also be great @Rafi993 when you complete this refactoring, if you could change the `alt` text on the profile image. In my opinion it should be empty since it's decorative in this context so `alt=""`.

Really great to see how this is developing - it's going to be a big improvement!